tests/templates/test_integration.py

```diff
@@ -647,3 +647,82 @@ def circuit():
 
         # Check that execution does not throw error
         circuit()
+
+
+class TestNonConsecutiveWires:
+    """Tests that a template results in the same state whether we use nonconsecutive wire labels or not.
+    """
+
+    @pytest.mark.parametrize("template, diffable, nondiffable, n_wires", QUBIT_DIFFABLE_NONDIFFABLE)
+    def test_qubit_result_is_wire_label_independent(self, template, diffable, nondiffable, n_wires):
+        """Tests integration of qubit templates with non-integer and non-consecutive wires."""
+
+        # merge differentiable and non-differentiable arguments
+        nondiffable.update(diffable)
+
+        # construct qnode with consecutive wires
+        dev_consec = qml.device('default.qubit', wires=n_wires)
+
+        @qml.qnode(dev_consec)
+        def circuit_consec():
+            template(**nondiffable)
+            return qml.expval(qml.Identity(wires=0))
+
+        # construct qnode with nonconsecutive wires
+        nonconsecutive_wires = np.random.choice(['a', 'b', 'c', 'd', 'e', 'f'], size=n_wires, replace=False)
+        kwargs_nonconsec = nondiffable.copy()
+        if 'wires' in kwargs_nonconsec:
+            kwargs_nonconsec['wires'] = nonconsecutive_wires
+        # DoubleExcitationLayers does not have a wires kwarg
+        if 'wires1' in kwargs_nonconsec:
+            kwargs_nonconsec['wires1'] = nonconsecutive_wires[:2]
+            kwargs_nonconsec['wires2'] = nonconsecutive_wires[2:]
+        # some kwargs in UCSSD need to be manually replaced
+        if 'ph' in kwargs_nonconsec:
+             kwargs_nonconsec['ph'] = [nonconsecutive_wires[:3], nonconsecutive_wires[1:]]
+        if 'pphh' in kwargs_nonconsec:
+             kwargs_nonconsec['pphh'] = [[nonconsecutive_wires[:2], nonconsecutive_wires[2:]]]
```

Comment:
This way of doing things is not very forward-compatible :crying_cat_face: (e.g., there could be future templates that satisfy these same conditions). Better to be explicit about testing different templates that require different conditions separately